src/util_i_targeting.nss

```diff
@@ -0,0 +1,894 @@
+                "WHERE sUUID = @sUUID " +
+                    "AND sVarName =@sVarName;";
+
+    sqlquery q = _PrepareTargetingQuery(s);
+    SqlBindString(q, "@sUUID", GetObjectUUID(oPC));
+    SqlBindString(q, "@sVarName", sVarName);
+
+    return SqlStep(q) ? SqlGetInt(q, 0) : 0;
+}
+
+string GetTargetingHookVarName(int nHookID)
+{
+    return _GetTargetingHookFieldData(nHookID, "sVarName");
+}
+
+int GetTargetingHookObjectType(int nHookID)
+{
+    return StringToInt(_GetTargetingHookFieldData(nHookID, "nObjectType"));
+}
+
+int GetTargetingHookUses(int nHookID)
+{
+    return StringToInt(_GetTargetingHookFieldData(nHookID, "nUses"));
+}
+
+string GetTargetingHookScript(int nHookID)
+{
+    return _GetTargetingHookFieldData(nHookID, "sScript");
+}
+
+int AddTargetingHook(object oPC, string sVarName, int nObjectType = OBJECT_TYPE_ALL,
+        string sScript = "", int nUses = 1)
+{
+    _CreateTargetingDataTables();
+
+    string s =  "REPLACE INTO targeting_hooks (sUUID, sVarName, nObjectType, nUses, sScript) " +
+                "VALUES (@sUUID, @sVarName, @nObjectType, @nUses, @sScript);";
+
+    sqlquery q = _PrepareTargetingQuery(s);
+    SqlBindString(q, "@sUUID", GetObjectUUID(oPC));
+    SqlBindString(q, "@sVarName", sVarName);
+    SqlBindInt   (q, "@nObjectType", nObjectType);
+    SqlBindInt   (q, "@nUses", nUses);
+    SqlBindString(q, "@sScript", sScript);
+    SqlStep(q);
+
+    if (IsDebugging(DEBUG_LEVEL_DEBUG))
+    {
+        Debug("Adding targeting hook ID " + HexColorString(IntToString(_GetLastTargetingHookID()), COLOR_CYAN) +
+            "\n  sVarName -> " + HexColorString(sVarName, COLOR_CYAN) +
+            "\n  nObjectType -> " + HexColorString(ObjectTypeToString(nObjectType), COLOR_CYAN) +
+            "\n  sScript -> " + (sScript == "" ? HexColorString("[None]", COLOR_RED_LIGHT) :
+                HexColorString(sScript, COLOR_CYAN)) +
+            "\n  nUses -> " + (nUses == -1 ? HexColorString("Unlimited", COLOR_CYAN) :
+                (nUses > 0 ? HexColorString(IntToString(nUses), COLOR_CYAN) :
+                HexColorString(IntToString(nUses), COLOR_RED_LIGHT))) + "\n");
+    }
+
+    return _GetLastTargetingHookID();
+}
+
+void DeleteTargetingHook(int nHookID)
+{
+    struct TargetingHook th = GetTargetingHookDataByHookID(nHookID);
+
+    if (th == TARGETING_HOOK_INVALID)
+    {
+        Warning("DeleteTargetingHook::Unable to retrieve valid targeting data for " +
+            "targeting hook " + IntToString(nHookID));
+        return;
+    }
+
+    if (IsDebugging(DEBUG_LEVEL_DEBUG))
+        Debug("Deleting targeting hook ID " + HexColorString(IntToString(nHookID), COLOR_CYAN) + "\n");
+
+    _DeleteTargetingHookData(nHookID);
+    DeleteLocalInt(th.oPC, TARGET_HOOK_ID);
+    DeleteLocalInt(th.oPC, TARGET_HOOK_BEHAVIOR);
+
+    if (th.sScript != "")
+    {
+        Debug("Running post-targeting script " + th.sScript);
+        RunTargetingHookScript(th.sScript, th.oPC);
```

Comment:
This will be resolved with the solution from the first comment.  

It should be internal.  I originally put it here because every hook is deleted when targeting is complete, and that's when the script is run. I can't think of a use case other than exiting targeting mode that the script run would be called, so a separate function likely isn't required, but it definitely shouldn't be part of a public function.